sycl/test/extensions/properties/new_properties.cpp

```diff
@@ -0,0 +1,207 @@
+constexpr properties pl2{pl1, naive{}};
+static_assert(pl1.template has_property<full_group>());
+static_assert(!pl1.template has_property<naive>());
+static_assert(pl2.template has_property<full_group>());
+static_assert(pl2.template has_property<naive>());
+
+enum class data_placement { blocked, striped };
+template <data_placement placement>
+struct data_placement_property
+    : named_property_base<data_placement_property<placement>,
+                          struct data_placement_property_key> {
+  static constexpr bool is_blocked() {
+    return placement == data_placement::blocked;
+  }
+};
+inline constexpr data_placement_property<data_placement::blocked> blocked;
+inline constexpr data_placement_property<data_placement::striped> striped;
+
+static_assert(properties{naive{}, blocked}
+                  .get_property<struct data_placement_property_key>()
+                  .is_blocked());
+static_assert(!properties{naive{}, striped}
+                   .get_property<struct data_placement_property_key>()
+                   .is_blocked());
+static_assert(
+    properties{naive{}, blocked}
+        .get_property_or_default_to<struct data_placement_property_key>(blocked)
+        .is_blocked());
+static_assert(
+    !properties{naive{}, data_placement_property<data_placement::striped>{}}
+         .get_property_or_default_to<struct data_placement_property_key>(
+             blocked)
+         .is_blocked());
+static_assert(
+    properties{naive{}}
+        .get_property_or_default_to<struct data_placement_property_key>(blocked)
+        .is_blocked());
+static_assert(
+    !properties{naive{}}
+         .get_property_or_default_to<struct data_placement_property_key>(
+             striped)
+         .is_blocked());
+
+constexpr properties pl3{full_group{}, blocked};
+// constexpr properties pl4{pl3, naive{}};
+template <typename... other_property_list_tys, typename... other_property_tys>
+constexpr auto merge_properties(
+    properties<detail::properties_type_list<other_property_list_tys...>>,
+    other_property_tys...) {
+  return 42;
+}
+static_assert(merge_properties(pl3, naive{}) == 42);
+} // namespace test_group_load_store
+
+namespace test_merge_ctor {
+template <int N> struct property : named_property_base<property<N>> {};
+
+constexpr properties pl1{property<1>{}, property<2>{}, property<3>{}};
+constexpr properties pl2{pl1, property<4>{}};
+} // namespace test_merge_ctor
+
+namespace test_compile_prop_in_runtime_list {
+template <int N>
+struct ct_prop : named_property_base<ct_prop<N>, struct ct_prop_key> {
+  static constexpr auto value() { return N; }
+};
+struct rt_prop : named_property_base<rt_prop> {
+  rt_prop(int N) : x(N) {}
+
+  int x;
+
+  constexpr auto value() { return x; }
+};
+void test() {
+  int x = 42;
+  properties pl{ct_prop<42>{}, rt_prop{x}};
+  constexpr auto p = pl.get_property<struct ct_prop_key>();
+  static_assert(std::is_same_v<decltype(p), const ct_prop<42>>);
```

Comment:
I'm not sure if `const ct_prop<42>` instead of `prop<42>` is an issue, and if it is then I don't know how to address it.